### ISSUE TITLE
perl debugger hangs after "|x @a"

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#8227]
Migrated from [rt.perl.org#37781](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=37781) (status was 'resolved')

Searchable as RT37781$

**Comment by p5pRT** [comment]
### From Ulrich.Windl@rz.uni-regensburg.de

#### Created by Ulrich.Windl@rz.uni-regensburg.de


This input will hang Perl\, making it unsresponsive to ^C while it uses
100% CPU&#8203;:

perl \-d \-e 1
@&#8203;a=\(1\.\.500\)
|x @&#8203;a
q

Pager in use is "less"\. "strace" is reporting repeated segmentation faults
like this&#8203;:
sigreturn\(\)                             = ? \(mask now \[INT\]\)
\-\-\- SIGSEGV \(Segmentation fault\) @&#8203; 0 \(0\) \-\-\-

Bug found in SUSE Linux releases 9\.2\, SLES9 SP2\, and 10\.0\. Novell/SUSE bug
id is is 135559\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

This perlbug was built using Perl v5.8.7 - Fri Sep  9 14:57:31 UTC 2005
It is being executed now by  Perl v5.8.7 - Fri Sep  9 14:52:52 UTC 2005.

Site configuration information for perl v5.8.7:

Configured by abuild at Fri Sep  9 14:52:52 UTC 2005.

Summary of my perl5 (revision 5 version 8 subversion 7) configuration:
  Platform:
    osname=linux, osvers=2.6.12, archname=i586-linux-thread-multi
    uname='linux smetana 2.6.12 #1 smp wed mar 23 21:52:37 utc 2005 i686 athlon i386 gnulinux '
    config_args='-ds -e -Dprefix=/usr -Dvendorprefix=/usr -Dinstallusrbinperl -Dusethreads -Di_db -Di_dbm -Di_ndbm -Di_gdbm -Duseshrplib=true -Doptimize=-O2 -march=i586 -mtune=i686 -fmessage-length=0 -Wall -D_FORTIFY_SOURCE=2 -g -Wall -pipe'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -DDEBUGGING -fno-strict-aliasing -pipe -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O2 -march=i586 -mtune=i686 -fmessage-length=0 -Wall -D_FORTIFY_SOURCE=2 -g -Wall -pipe',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DTHREADS_HAVE_PIDS -DDEBUGGING -fno-strict-aliasing -pipe'
    ccversion='', gccversion='4.0.2 20050901 (prerelease) (SUSE Linux)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =''
    libpth=/lib /usr/lib /usr/local/lib
    libs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    perllibs=-lnsl -ldl -lm -lcrypt -lutil -lpthread -lc
    libc=/lib/libc-2.3.5.so, so=so, useshrplib=true, libperl=libperl.so
    gnulibc_version='2.3.5'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-Wl,-E -Wl,-rpath,/usr/lib/perl5/5.8.7/i586-linux-thread-multi/CORE'
    cccdlflags='-fPIC', lddlflags='-shared'

Locally applied patches:
    


@INC for perl v5.8.7:
    /usr/lib/perl5/5.8.7/i586-linux-thread-multi
    /usr/lib/perl5/5.8.7
    /usr/lib/perl5/site_perl/5.8.7/i586-linux-thread-multi
    /usr/lib/perl5/site_perl/5.8.7
    /usr/lib/perl5/site_perl
    /usr/lib/perl5/vendor_perl/5.8.7/i586-linux-thread-multi
    /usr/lib/perl5/vendor_perl/5.8.7
    /usr/lib/perl5/vendor_perl
    .


Environment for perl v5.8.7:
    HOME=/home/wiu09524
    LANG=de_DE.UTF-8
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/usr/local/bin:/usr/bin:/usr/X11R6/bin:/bin:/usr/games:/opt/gnome/bin:/opt/kde3/bin:/usr/lib/jvm/jre/bin:/usr/lib/mit/bin:/usr/lib/mit/sbin
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @richardfoley

This is a well known bug which jumps up and bites people every now and then\.

Michaels fix may well be the answer&#8203;:
&nbsp;    
&nbsp;    Message\-ID&#8203;: \<20051128185434\.GA20579@&#8203;immd4\.informatik\.uni\-erlangen\.de>
&nbsp;    Subject&#8203;: IO redirection problems in the debugger
&nbsp;    From&#8203;: Michael Schroeder \<Michael\.Schroeder@&#8203;informatik\.uni\-erlangen\.de>

It may also be addressed by setting the PERL\_RL=1 variable to non\-zero\, as 
described in this email&#8203;:

&nbsp;    Message\-ID&#8203;: \<1DzpuX\-0SEXLM0@&#8203;fwd24\.aul\.t\-online\.de>
&nbsp;    Subject&#8203;: Re&#8203;: Re&#8203;: \[perl \#22758\] debugger command "|x" swallows all output 
&nbsp;    From&#8203;: "Richard\.Foley@&#8203;t\-online\.de" \<Richard\.Foley@&#8203;t\-online\.de>

Also mentioned here&#8203;:
&nbsp;    
&nbsp;    Message\-ID&#8203;: \<42EE0858\.16850\.C9E632@&#8203;rkdvmks1\.ngate\.uni\-regensburg\.de>
&nbsp;    Re&#8203;: \[perl \#22758\] debugger command "|x" swallows all output
&nbsp;    From&#8203;: "Ulrich Windl" \<ulrich\.windl@&#8203;rz\.uni\-regensburg\.de>

Cheers\.

\-\- 

Ciao
Richard Foley

http&#8203;://www\.oreilly\.com/catalog/perldebugpr/index\.html
http&#8203;://www\.apress\.com/book/bookDisplay\.html?bID=399
http&#8203;://www\.oreilly\.com/catalog/rtessentials/index\.html

On Tuesday 29 November 2005 09&#8203;:47\, Ulrich Windl wrote&#8203;:
> \# New Ticket Created by  "Ulrich Windl"
> \# Please include the string&#8203;:  \[perl \#37781\]
> \# in the subject line of all future correspondence about this issue\.
> \# \<URL&#8203;: https://rt-archive.perl.org/perl5/Ticket/Display.html?id=37781 >
>
>
>
> This is a bug report for perl from Ulrich\.Windl@&#8203;rz\.uni\-regensburg\.de\,
> generated with the help of perlbug 1\.35 running under perl v5\.8\.7\.
>
>
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> This input will hang Perl\, making it unsresponsive to ^C while it uses
> 100% CPU&#8203;:
>
> perl \-d \-e 1
> @&#8203;a=\(1\.\.500\)
>
> |x @&#8203;a
>
> q
>
> Pager in use is "less"\. "strace" is reporting repeated segmentation faults
> like this&#8203;:
> sigreturn\(\)                             = ? \(mask now \[INT\]\)
> \-\-\- SIGSEGV \(Segmentation fault\) @&#8203; 0 \(0\) \-\-\-
>
> Bug found in SUSE Linux releases 9\.2\, SLES9 SP2\, and 10\.0\. Novell/SUSE bug
> id is is 135559\.
> \[Please do not change anything below this line\]
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \-\-\-
> Flags&#8203;:
>     category=core
>     severity=medium
> \-\-\-
> This perlbug was built using Perl v5\.8\.7 \- Fri Sep  9 14&#8203;:57&#8203;:31 UTC 2005
> It is being executed now by  Perl v5\.8\.7 \- Fri Sep  9 14&#8203;:52&#8203;:52 UTC 2005\.
>
> Site configuration information for perl v5\.8\.7&#8203;:
>
> Configured by abuild at Fri Sep  9 14&#8203;:52&#8203;:52 UTC 2005\.
>
> Summary of my perl5 \(revision 5 version 8 subversion 7\) configuration&#8203;:
>   Platform&#8203;:
>     osname=linux\, osvers=2\.6\.12\, archname=i586\-linux\-thread\-multi
>     uname='linux smetana 2\.6\.12 \#1 smp wed mar 23 21&#8203;:52&#8203;:37 utc 2005 i686
> athlon i386 gnulinux ' config\_args='\-ds \-e \-Dprefix=/usr
> \-Dvendorprefix=/usr \-Dinstallusrbinperl \-Dusethreads \-Di\_db \-Di\_dbm
> \-Di\_ndbm \-Di\_gdbm \-Duseshrplib=true \-Doptimize=\-O2 \-march=i586 \-mtune=i686
> \-fmessage\-length=0 \-Wall \-D\_FORTIFY\_SOURCE=2 \-g \-Wall \-pipe'
> hint=recommended\, useposix=true\, d\_sigaction=define
>     usethreads=define use5005threads=undef useithreads=define
> usemultiplicity=define useperlio=define d\_sfio=undef uselargefiles=define
> usesocks=undef use64bitint=undef use64bitall=undef uselongdouble=undef
>     usemymalloc=n\, bincompat5005=undef
>   Compiler&#8203;:
>     cc='cc'\, ccflags ='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DTHREADS\_HAVE\_PIDS
> \-DDEBUGGING \-fno\-strict\-aliasing \-pipe \-D\_LARGEFILE\_SOURCE
> \-D\_FILE\_OFFSET\_BITS=64'\, optimize='\-O2 \-march=i586 \-mtune=i686
> \-fmessage\-length=0 \-Wall \-D\_FORTIFY\_SOURCE=2 \-g \-Wall \-pipe'\,
> cppflags='\-D\_REENTRANT \-D\_GNU\_SOURCE \-DTHREADS\_HAVE\_PIDS \-DDEBUGGING
> \-fno\-strict\-aliasing \-pipe' ccversion=''\, gccversion='4\.0\.2 20050901
> \(prerelease\) \(SUSE Linux\)'\, gccosandvers='' intsize=4\, longsize=4\,
> ptrsize=4\, doublesize=8\, byteorder=1234 d\_longlong=define\, longlongsize=8\,
> d\_longdbl=define\, longdblsize=12 ivtype='long'\, ivsize=4\, nvtype='double'\,
> nvsize=8\, Off\_t='off\_t'\, lseeksize=8 alignbytes=4\, prototype=define
>   Linker and Libraries&#8203;:
>     ld='cc'\, ldflags =''
>     libpth=/lib /usr/lib /usr/local/lib
>     libs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
>     perllibs=\-lnsl \-ldl \-lm \-lcrypt \-lutil \-lpthread \-lc
>     libc=/lib/libc\-2\.3\.5\.so\, so=so\, useshrplib=true\, libperl=libperl\.so
>     gnulibc\_version='2\.3\.5'
>   Dynamic Linking&#8203;:
>     dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-Wl\,\-E
> \-Wl\,\-rpath\,/usr/lib/perl5/5\.8\.7/i586\-linux\-thread\-multi/CORE'
> cccdlflags='\-fPIC'\, lddlflags='\-shared'
>
> Locally applied patches&#8203;:
>
>
> \-\-\-
> @&#8203;INC for perl v5\.8\.7&#8203;:
>     /usr/lib/perl5/5\.8\.7/i586\-linux\-thread\-multi
>     /usr/lib/perl5/5\.8\.7
>     /usr/lib/perl5/site\_perl/5\.8\.7/i586\-linux\-thread\-multi
>     /usr/lib/perl5/site\_perl/5\.8\.7
>     /usr/lib/perl5/site\_perl
>     /usr/lib/perl5/vendor\_perl/5\.8\.7/i586\-linux\-thread\-multi
>     /usr/lib/perl5/vendor\_perl/5\.8\.7
>     /usr/lib/perl5/vendor\_perl
>     \.
>
> \-\-\-
> Environment for perl v5\.8\.7&#8203;:
>     HOME=/home/wiu09524
>     LANG=de\_DE\.UTF\-8
>     LANGUAGE \(unset\)
>     LD\_LIBRARY\_PATH \(unset\)
>     LOGDIR \(unset\)
>    
> PATH=/usr/local/bin&#8203;:/usr/bin&#8203;:/usr/X11R6/bin&#8203;:/bin&#8203;:/usr/games&#8203;:/opt/gnome/bin&#8203;:
>/opt/kde3/bin&#8203;:/usr/lib/jvm/jre/bin&#8203;:/usr/lib/mit/bin&#8203;:/usr/lib/mit/sbin
> PERL\_BADLANG \(unset\)
>     SHELL=/bin/bash

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @kevindawson

see \#22758

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

see \#22758

**Comment by p5pRT** [comment]
### From @jkeenan

On Fri Aug 23 09&#8203;:20&#8203;:51 2013\, bowtie wrote&#8203;:
> see \#22758

Closing\, as problem appears to be resolved as noted in RT \#22758\.

Thank you very much\.
Jim Keenan

**Comment by p5pRT** [comment]
@jkeenan - Status changed from 'open' to 'resolved'
